sys/kern/src/arch/rv64/mtimer.rs

```diff
@@ -29,16 +29,28 @@ const MTIME: u64 = 0x0200_BFF8;
 //
 #[no_mangle]
 pub unsafe fn set_timer(tick_divisor: u32) {
-    // Set high-order bits of mtime to zero.  We only call this function prior
-    // to enabling interrupts so it should be safe.
-    unsafe {
-        core::ptr::write_volatile(MTIME as *mut u64, 0);
-        core::ptr::write_volatile(MTIMECMP as *mut u64, tick_divisor.into());
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "riscv-supervisor-mode")] {
+            // No access to mtimer for supervisor mode
```

Comment:
Should this be reading from stime?

Reply:
We will need to check for Sstc support on the platform and determine if we should set stimecmp directly or make a SBI call into M-mode. I think Tom is planning to add that as a follow-up